compiler/rustc_const_eval/src/transform/validate.rs

```diff
@@ -573,21 +573,7 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
                     _ => self.fail(location, format!("There is no checked version of {:?}", op)),
                 }
             }
-            Rvalue::UnaryOp(op, operand) => {
-                let a = operand.ty(&self.body.local_decls, self.tcx);
-                match op {
-                    UnOp::Neg => {
-                        check_kinds!(a, "Cannot negate type {:?}", ty::Int(..) | ty::Float(..))
-                    }
-                    UnOp::Not => {
-                        check_kinds!(
-                            a,
-                            "Cannot binary not type {:?}",
-                            ty::Int(..) | ty::Uint(..) | ty::Bool
-                        );
-                    }
-                }
-            }
+            Rvalue::UnaryOp(..) => {}
```

Comment:
This validation should not be removed without at least a valid explanation for why it's not necessary anymore :sweat:

Reply:
cc @JakobDegen who added this validation

Reply:
It's too late last night, this should be a Draft.
I just found more document for this validations, https://github.com/rust-lang/rust/pull/95966/commits/634369170a5bc73daa51b2b15763eb508885d9c0